Fix text color for Compose theme in night mode

ovitrif opened the pull request:
Fixes #17085

> The implementation applies the suggestion from here: https://github.com/wordpress-mobile/WordPress-Android/pull/16934#discussion_r929884451.

To test:

0. Make sure the device is set to night mode
1. Open Jetpack app and Login if you're not already logged in
3. Tap `Me`
4. Tap `Scan Login Code`
5. In a guest Chrome session on a computer, go to www.wordpress.com and 
   - Tap `Log In`
   - Tap `Login via the mobile app`
6. Scan the QR Code with your device
7. **Expect** the title text to be colored white → making the text readable.

## Regression Notes
1. Potential unintended areas of impact
  Title text color in `day mode` on the screens after scanning the login code.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
  Manual testing & compose preview.

3. What automated tests I added (or what prevented me from doing so)
  No tests warranted by UI tweaks.


### Previews

| Before | After |
| --- | --- |
| ![d_old](https://user-images.githubusercontent.com/4588074/186404599-92cd85c0-7c35-411f-9381-d1729dba23b1.jpg) | ![d_new](https://user-images.githubusercontent.com/4588074/186404613-3826e9a7-85d5-415f-a394-bbd59c5e0f99.jpg) |


<details>
<summary>Day mode unchanged</summary>

| <img width="314" src="https://user-images.githubusercontent.com/4588074/186404772-84ce2697-e416-4ebc-bcbb-200f75fea970.jpg"> |
| --- |

No regression here :shipit:.
</details>

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ovitrif committed:
Add compose AppThemeWithBackground to be used for most layouts

ovitrif committed:
Use compose AppThemeWithBackground for QrCodeAuthScreen

ovitrif committed:
Use compose AppThemeWithBackground for previews of QrCodeAuthScreen composables

ovitrif committed:
Update theme function naming to suggest using the app theme with surface

mkevins commented on WordPress/src/main/java/org/wordpress/android/ui/compose/theme/AppThemeWihoutBackground.kt:
```diff
@@ -11,22 +11,22 @@ import org.wordpress.android.BuildConfig
 @Composable
-fun AppThemeWithBackground(
+fun AppTheme(
     isDarkTheme: Boolean = isSystemInDarkTheme(),
     content: @Composable () -> Unit
 ) {
-    AppTheme(isDarkTheme) {
+    AppThemeWithoutBackground(isDarkTheme) {
```
Nice 👍 . I like the semantics of this when leaving public the theme without surface. Also, the doc comments are informative!

ovitrif replied:
Team work 🎸 🚀

Commit:
Fix filename and add preview

This is a small suggestion commit to fix a filename typo (revert to the
simpler name), to use a dark-mode preview, and to fix a few lint issues.

mkevins reviewed (approved):
Thanks for fixing this Ovi! I tested this out on a Pixel 3a (physical device) and it is working as described. The code changes also look good.

I've added a suggestion commit on this branch:

https://github.com/wordpress-mobile/WordPress-Android/compare/issue/17085-compose-theme-night-mode-text-color...issue/17085-compose-theme-night-mode-text-color--suggestion

to fix a filename typo, and some mint lint issues (also adding a dark-mode preview for the content). If it's useful, feel free to fast forward this branch to that one before merging 👍 (I've tested both).

mkevins commented:
Oh, a reminder to add the release note, too, since this is already in production.

ovitrif committed:
Update release notes

ovitrif committed:
Merge branch 'trunk' into issue/17085-compose-theme-night-mode-text-color

ovitrif committed:
Update release notes after merge

ovitrif committed:
Improve docs for app theme composables

ovitrif commented:
> I've added a suggestion commit on this branch:
> 
> [issue/17085-compose-theme-night-mode-text-color...issue/17085-compose-theme-night-mode-text-color--suggestion](https://github.com/wordpress-mobile/WordPress-Android/compare/issue/17085-compose-theme-night-mode-text-color...issue/17085-compose-theme-night-mode-text-color--suggestion)
> 
> to fix a filename typo, and some mint lint issues (also adding a dark-mode preview for the content). If it's useful, feel free to fast forward this branch to that one before merging 👍 (I've tested both).

Thank you for this @mkevins, that's really awesome ❤️

I added your suggestions to the branch of this PR 🎉
